tests/unit/asset_tests.cpp

```diff
@@ -491,6 +491,84 @@ EOSIO_TEST_BEGIN(extended_asset_type_test)
+   CHECK_EQUAL( ( 0LL * extended_asset{ asset{ 0LL, sym_no_prec}, {}} ), (extended_asset{ asset{ 0LL, sym_no_prec}, {}}) );
+   CHECK_EQUAL( ( 1LL * extended_asset{ asset{ 2LL, sym_no_prec}, {}} ), (extended_asset{ asset{ 2LL, sym_no_prec}, {}}) );
+   CHECK_EQUAL( (-1LL * extended_asset{ asset{ 2LL, sym_no_prec}, {}} ), (extended_asset{ asset{-2LL, sym_no_prec}, {}}) );
+
+   CHECK_ASSERT( "multiplication underflow", (
+      [&]() {
+         2LL * extended_asset{ asset{asset_min, sym_no_prec}, {}};
+      })
+   )
+
+   CHECK_ASSERT( "multiplication overflow", (
+      [&]() {
+         2LL * extended_asset{ asset{ asset_max, sym_no_prec}, {}};
+      })
+   )
+
+   // --------------------------------------------------------------------------
+   // friend int64_t operator/( const extended_asset& a, const extended_asset& b )
+   CHECK_EQUAL( (extended_asset{ asset{ 0LL, sym_no_prec}, {}} / extended_asset{ asset{ 1LL, sym_no_prec}, {}}), 0LL )
+   CHECK_EQUAL( (extended_asset{ asset{ 1LL, sym_no_prec}, {}} / extended_asset{ asset{ 1LL, sym_no_prec}, {}}), 1LL )
+   CHECK_EQUAL( (extended_asset{ asset{ 4LL, sym_no_prec}, {}} / extended_asset{ asset{ 2LL, sym_no_prec}, {}}), 2LL )
+   CHECK_EQUAL( (extended_asset{ asset{-4LL, sym_no_prec}, {}} / extended_asset{ asset{ 2LL, sym_no_prec}, {}}), -2LL )
+   CHECK_EQUAL( (extended_asset{ asset{-4LL, sym_no_prec}, {}} / extended_asset{ asset{-2LL, sym_no_prec}, {}}), 2LL )
+
+   // --------------------------------------------------------------------------
+   // friend extended_asset& operator/=( extended_asset& a, int64_t b )
+   CHECK_EQUAL( (extended_asset{ asset{ 0LL, sym_no_prec}, {}} /= 1LL ), (extended_asset{ asset{ 0LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{ 1LL, sym_no_prec}, {}} /= 1LL ), (extended_asset{ asset{ 1LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{ 4LL, sym_no_prec}, {}} /= 2LL ), (extended_asset{ asset{ 2LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{-4LL, sym_no_prec}, {}} /= 2LL ), (extended_asset{ asset{-2LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{-4LL, sym_no_prec}, {}} /= -2LL ), (extended_asset{ asset{ 2LL, sym_no_prec}, {}}) )
+
+   // --------------------------------------------------------------------------
+   // friend extended_asset operator/( const extended_asset& a, int64_t b )
+   CHECK_EQUAL( (extended_asset{ asset{ 0LL, sym_no_prec}, {}} / 1LL ), (extended_asset{ asset{ 0LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{ 1LL, sym_no_prec}, {}} / 1LL ), (extended_asset{ asset{ 1LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{ 4LL, sym_no_prec}, {}} / 2LL ), (extended_asset{ asset{ 2LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{-4LL, sym_no_prec}, {}} / 2LL ), (extended_asset{ asset{-2LL, sym_no_prec}, {}}) )
+   CHECK_EQUAL( (extended_asset{ asset{-4LL, sym_no_prec}, {}} / -2LL ), (extended_asset{ asset{ 2LL, sym_no_prec}, {}}) )
+
```

Comment:
```suggestion
    auto a0 = asset{ 0LL, sym_no_prec};
   auto a1 = asset{ 1LL, sym_no_prec};
   auto a2 = asset{ 2LL, sym_no_prec};
   auto a4 = asset{ 4LL, sym_no_prec};
   auto aneg1 = asset{ -1LL, sym_no_prec};
   auto aneg2 = asset{ -2LL, sym_no_prec};
   auto aneg4 = asset{ -4LL, sym_no_prec};

   // --------------------------------------------------------------------------
   // extended_asset& operator*=( int64_t b )
   CHECK_EQUAL( (extended_asset{ a0, {}} *=  0LL ), (extended_asset{ a0, {}}) );
   CHECK_EQUAL( (extended_asset{ a2, {}} *=  1LL ), (extended_asset{ a2, {}}) );
   CHECK_EQUAL( (extended_asset{ a2, {}} *= -1LL ), (extended_asset{ aneg2, {}}) );

   CHECK_ASSERT( "multiplication underflow", (
      [&]() {
         extended_asset{ asset{asset_min, sym_no_prec}, {}} *= 2LL;
      })
   )

   CHECK_ASSERT( "multiplication overflow", (
      [&]() {
         extended_asset{ asset{ asset_max, sym_no_prec}, {}} *= 2LL;
      })
   )

   // --------------------------------------------------------------------------
   // friend extended_asset operator*( const extended_asset& a, int64_t b )
   CHECK_EQUAL( (extended_asset{ a0, {}} *  0LL ), (extended_asset{ a0, {}}) );
   CHECK_EQUAL( (extended_asset{ a2, {}} *  1LL ), (extended_asset{ a2, {}}) );
   CHECK_EQUAL( (extended_asset{ a2, {}} * -1LL ), (extended_asset{ aneg2, {}}) );

   CHECK_ASSERT( "multiplication underflow", (
      [&]() {
         extended_asset{ asset{asset_min, sym_no_prec}, {}} * 2LL;
      })
   )

   CHECK_ASSERT( "multiplication overflow", (
      [&]() {
         extended_asset{ asset{ asset_max, sym_no_prec}, {}} * 2LL;
      })
   )

   // --------------------------------------------------------------------------
   // friend extended_asset operator*( int64_t a, const extended_asset& b )
   CHECK_EQUAL( ( 0LL * extended_asset{ a0, {}} ), (extended_asset{ a0, {}}) );
   CHECK_EQUAL( ( 1LL * extended_asset{ a2, {}} ), (extended_asset{ a2, {}}) );
   CHECK_EQUAL( (-1LL * extended_asset{ a2, {}} ), (extended_asset{ aneg2, {}}) );

   CHECK_ASSERT( "multiplication underflow", (
      [&]() {
         2LL * extended_asset{ asset{asset_min, sym_no_prec}, {}};
      })
   )

   CHECK_ASSERT( "multiplication overflow", (
      [&]() {
         2LL * extended_asset{ asset{ asset_max, sym_no_prec}, {}};
      })
   )

   // --------------------------------------------------------------------------
   // friend int64_t operator/( const extended_asset& a, const extended_asset& b )
   CHECK_EQUAL( (extended_asset{ a0, {}} / extended_asset{ a1, {}}), 0LL )
   CHECK_EQUAL( (extended_asset{ a1, {}} / extended_asset{ a1, {}}), 1LL )
   CHECK_EQUAL( (extended_asset{ a4, {}} / extended_asset{ a2, {}}), 2LL )
   CHECK_EQUAL( (extended_asset{ aneg4, {}} / extended_asset{ a2, {}}), -2LL )
   CHECK_EQUAL( (extended_asset{ aneg4, {}} / extended_asset{ aneg2, {}}), 2LL )

   // --------------------------------------------------------------------------
   // friend extended_asset& operator/=( extended_asset& a, int64_t b )
   CHECK_EQUAL( (extended_asset{ a0, {}} /= 1LL ), (extended_asset{ a0, {}}) )
   CHECK_EQUAL( (extended_asset{ a1, {}} /= 1LL ), (extended_asset{ a1, {}}) )
   CHECK_EQUAL( (extended_asset{ a4, {}} /= 2LL ), (extended_asset{ a2, {}}) )
   CHECK_EQUAL( (extended_asset{ aneg4, {}} /= 2LL ), (extended_asset{ aneg2, {}}) )
   CHECK_EQUAL( (extended_asset{ aneg4, {}} /= -2LL ), (extended_asset{ a2, {}}) )

```

What do you think of this simplification?

Reply:
See 84d75fe0 for a change based on this suggestion.

Reply:
Nice!